File: trap.jl
function deltax(a, b, n)
    ac = b - a
    mc = ac / n
    return mc
end

function func(a)
    # equation a^2 + 1 here
    x = a * a + 1
    return x
end


function faloop(a, b, n, m)
    inside = 0
    
    counter = a + m
    while counter < b
        inside += 2*func(counter)
        counter += m
    end
    fnl = inside + func(a) + func(b)
    return fnl
end


function askdata()
    # low bound
    a = 0.0
    # upper bound
    b = 3.0
    # number of subintervals
    n = 500000000.0
    m = deltax(a, b, n)
    final = m/2 * faloop(a, b, n, m)

    return final
end


final = askdata()

println(final)

File: trap.c
#include <stdio.h>
#include <math.h>

float deltax(float a, float b, float n){

    float ac = b - a;
    float mc = ac / n;
    return mc;
}

float func(float a){
    //equation a^2 + 1 here
    float x = a * a + 1;
    return x;
}

float faloop(float a, float b, float n, float m){
    float inside = 0;
    
    float counter = a + m;
    while(counter < b) {
        inside += 2*func(counter);
        counter += m;
    }
    float fnl = inside + func(a) + func(b);
    return fnl;
}

float askdata(){
    //low bound
    float a = 0.0;
    //upper bound
    float b = 3.0;
    //number of subintervals
    float n = 5000.0;
    float m = deltax(a, b, n);
    float final = m/2 * faloop(a, b, n, m);
    return final;
}

float main(){
    float final = askdata();

    printf("%f", final);
    return 0;

}